tests/parametric/test_otel_span_methods.py

```diff
@@ -396,6 +398,212 @@ def test_otel_set_attributes_separately(self, test_agent, test_library):
             assert span["resource"] == "operation"
 
+    @missing_feature(context.library < "java@1.26.0", reason="Implemented in 1.26.0")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_started_with_link_from_another_span(self, test_agent, test_library):
+        """Test adding a span link created from another span.
+        This tests the functionality of "create a direct link between two spans
+        given two valid span (or SpanContext) objects" as specified in the RFC.
+        """
+        with test_library:
+            with test_library.otel_start_span("root") as parent:
+                parent.end_span()
+                with test_library.otel_start_span(
+                    "child",
+                    parent_id=parent.span_id,
+                    links=[Link(parent_id=parent.span_id, attributes={"foo": "bar", "array": ["a", "b", "c"]})],
+                ) as child:
+                    child.end_span()
+
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, otel_span(name="root"))
+        assert len(trace) == 2
+
+        root = find_span(trace, otel_span(name="root"))
+        child = find_span(trace, otel_span(name="child"))
+        assert child.get("parent_id") == root.get("span_id")
+
+        span_links = retrieve_span_links(child)
+        assert span_links is not None
+        assert len(span_links) == 1
+
+        link = span_links[0]
+        assert link.get("span_id") == root.get("span_id")
+        assert link.get("trace_id") == root.get("trace_id")
+        root_tid = root["meta"].get("_dd.p.tid") or "0" if "meta" in root else "0"
+        assert (link.get("trace_id_high") or 0) == int(root_tid, 16)
+        assert link["attributes"].get("foo") == "bar"
+        assert link["attributes"].get("array.0") == "a"
+        assert link["attributes"].get("array.1") == "b"
+        assert link["attributes"].get("array.2") == "c"
+        # TODO Why does flags is expected to be 0, meaning "not set"? 
+        # If 0, it is not sampled so it won't be send to the agent.
+        # assert (link.get("flags") or 0) == 0
+
+    @missing_feature(context.library < "java@1.26.0", reason="Implemented in 1.26.0")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_started_with_link_from_datadog_headers(self, test_agent, test_library):
+        """Properly inject datadog distributed tracing information into span links.
```

Comment:
Should we always generate span links from distributed trace headers? 

In the python tracer we only generate a span link if conflicting spanids/traceids are found in http headers. In all other cases `x-datadog-parent-id` is only used to set parent-child relationships.

Reply:
Ahhh we'll need to test this behavior (from the RFC): 

```
When continuing a distributed trace, the context having extracted it must add a span link referencing the active span in the extraction context, if present, to any span which the distributed tracing context is applied to. (For example, when there is an existing local trace and then an incoming request comes in.)
Informational: This ensures that it will be possible to find the true source context for a span.
Example
Given:
The current active trace context is T1/S1 (TraceID: 1, SpanID: 1),
An incoming request arrives with the trace context T2/S2,
The tracer extracts the incoming trace context and starts a new span (T2/S3) with the extracted incoming context.
```